lib/salus/processor.rb

```diff
@@ -96,13 +96,14 @@ def fetch_config_file(source_uri, repo_path)
 
     def run_scanner(config, scanner_class, scanner_name)
       threads = []
+      reporting_threads = []
       files_copied = []
       reraise_exceptions = ENV.key?('RUNNING_SALUS_TESTS')
 
       # We won't auto cleanup - we'll manually delete any copied fields after
       # the various threads have finished
       copied = RepoSearcher.new(@repo_path, config, false).matching_repos do |repo|
-        threads << Thread.new do
+        thread = Thread.new do
```

Comment:
This works.  I could see a case where threads becoming scanning threads but as written what you have will work.
```
if canner_instance.is_reporting_scanner?
  reporting_threads << thread
else
  scanning_threads << thread
end
```